packages/sdk-core/src/common/DelayHelper.ts

```diff
@@ -6,12 +8,21 @@ export class Delay {
      */
     public static wait(timeout: number, signal?: AbortSignal) {
         return new Promise<void>((resolve, reject) => {
+            // intervalId has to be defined here,
+            // as abortCallback can execute before setTimeout is executed
+            // eslint-disable-next-line prefer-const
+            let intervalId: NodeJS.Timeout | undefined;
+
             function abortCallback() {
                 clearTimeout(intervalId);
-                reject(new Error('Operation cancelled.'));
+                reject(new AbortError());
```

Comment:
I think this can cause an unhandled promise rejection that we can throw on the app exit. Because of that the app can exit with a different exit code. Am I Correct?

Reply:
Yes, you have to handle `delay` exceptions in case of aborting. I do this in `BacktraceMetrics`.